src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts

```diff
@@ -640,13 +640,12 @@ export class NotebookService extends Disposable implements INotebookService {
 			providerDisplayName: data.providerDisplayName,
 			exclusive: data.exclusive,
 			priority: RegisteredEditorPriority.default,
-			selectors: [],
-			externalEditor: !!data.externalEditor
+			selectors: []
 		});
 
 		info.update({ selectors: data.filenamePattern });
 
-		const reg = this.notebookProviderInfoStore.add(info);
+		const reg = this.notebookProviderInfoStore.addNotebookInfo(info);
```

Comment:
We still want to call `this.notebookProviderInfoStore.registerEditorForNotebook` in `mainThreadNotebook#$registerNotebookSerializer`, right?

Reply:
For example, Live Share registers serializers through code only, which is only calling `$registerNotebookSerializer`

Reply:
ah, I didn't notice that special case. An alternative is just to check the notebook provider is built-in every time and don't register a notebook editor if so. https://github.com/microsoft/vscode/pull/187070

We already are doing this when restoring info from the momento, so why not just make it a general check?